### PR TITLE
pallet-treasury skeleton

### PR DESCRIPTION
**Pull request by chiefbiiko**
## Summary
Checking in `pallet-treasury` boilerplate for further iterations (renamed to accommodate for more monetary mechanics pending).

### CODE REVIEW

File: pallets/treasury/src/inflation.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+pub struct InflationInfo {
+    pub(crate) annual: Range<Perbill>,
+    pub(crate) round: Range<Perbill>,
+    pub(crate) rewards_alloc: RewardsAllocationConfig,
+}
+
+impl InflationInfo {
+    /// Reset round inflation rate based on changes to round term.
+    pub fn update_round_term(&mut self, new_round_term: u32) {
+        let periods = BLOCKS_PER_YEAR / new_round_term;
```
Comment (AwesomeIbex) [inline pallets/treasury/src/inflation.rs:62]: might wanna ensure divice by > 0

**Review by AwesomeIbex — DISMISSED** [review]
Happy with this as a stub

File: .github/workflows/circuit.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
       - name: ⚙️Get nightly rust toolchain with wasm target
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly-2021-11-01
+          toolchain: nightly-2022-06-17
```
[the remark this answers is not in the document]
Comment (AwesomeIbex) [reply .github/workflows/circuit.yml:27]: > whats wrong here?

you don't have to duplicate it across the 2x toolchains in the other workflow, and here.

[the remark this answers is not in the document]
Comment (chiefbiiko) [reply .github/workflows/circuit.yml:27]: cool lets go ahead and introduce such standard file
would just outsource it to some other pr 🔜
needed this update here 4 the green light🚦

**Review by AwesomeIbex — DISMISSED** [review]

**Pull request closed without merging**